packages/myst-frontmatter/src/socials/validators.ts

```diff
@@ -0,0 +1,224 @@
+const YOUTUBE_REGEX = /^@?([A-Z0-9.]+)$/i;
+// Match a basic identifier (letters, numbers, full-stop, underscores)
+const THREADS_REGEX = /^[A-Z0-9_.]+$/i;
+// Match a basic identifier (letters, numbers, underscores, between 4 and 15 characters)
+const TWITTER_REGEX = /^@?([A-Z0-9_]{4,15})$/i;
+const TWITTER_URL_REGEX = /^https:\/\/(twitter\.com|x\.com)\/@?([A-Z0-9_]{4,15})$/;
+// Match a basic identifier (letters, numbers, underscores, full-stops)
+const GITHUB_USERNAME_REGEX = /^@?([A-Z0-9_.-]+)$/i;
+const GITHUB_ORG_URL_REGEX = /^https:\/github\.com\/orgs\/[A-Z0-9_.-]+$/;
+
+/**
+ * Validate value is valid Mastodon webfinger account
+ *
+ * See https://docs.joinmastodon.org/spec/webfinger/ for details on how
+ * @foo@server.com account URIs are resolved
+ */
+export function validateMastodon(input: any, opts: ValidationOptions) {
+  const value = validateString(input, opts);
+  if (value === undefined) return undefined;
+
+  const match = value.match(MASTODON_REGEX);
+  if (match === null)
+    return validationError(
+      `must be a user ID of the form @username@server e.g. @mystmarkdown@fosstodon.org`,
+      opts,
+    );
+
+  const username = match[1];
+  const host = validateDomain(match[2], opts);
+  if (host === undefined) return undefined;
+  return `@${username}@${host}`;
+}
+
+/**
+ * Validate value is valid Bluesky URI or URL string
+ */
+export function validateBluesky(input: any, opts: ValidationOptions) {
+  const value = validateString(input, opts);
+  if (value === undefined) return undefined;
+
+  // Username
+  let match: ReturnType<typeof value.match>;
+  if ((match = value.match(BLUESKY_REGEX))) {
+    const domain = validateDomain(match[1], opts);
+    if (domain === undefined) return undefined;
+    return `@${domain}`;
+  }
+  // URL to profile
+  else {
+    const result = validateUrl(value, opts);
+    if (result === undefined) return undefined;
+
+    match = result.match(BLUESKY_URL_REGEX);
+    if (match === null) {
+      return validationError(
+        `Bluesky profile URL must be a valid URL starting with https://bsky.app/profile/: ${value}`,
+        opts,
+      );
+    }
+
+    const domain = validateDomain(match[1], opts);
+    if (domain === undefined) return undefined;
+
+    return `@${domain}`;
+  }
+}
+
+/**
+ * Validate value is valid Twitter/X URL or username string
+ */
+export function validateTwitter(input: any, opts: ValidationOptions) {
+  const value = validateString(input, opts);
+  if (value === undefined) return undefined;
+  let match: ReturnType<typeof value.match>;
+  // URL
+  if ((match = value.match(TWITTER_URL_REGEX))) {
+    return match[1];
+  }
+  // username
+  else if ((match = value.match(TWITTER_REGEX))) {
+    return match[1];
+  } else {
+    return validationError(
+      `Twitter social identity must be a valid URL starting with https://twitter.com/, https://x.com/, or a valid username: ${value}`,
+      opts,
+    );
+  }
+}
+
```

Comment:
Twitter usernames are _not_ themselves `@foo`, they're `foo`. But people are used to typing `@foo`, so we can strip that off.